### PR TITLE
[examples][CI] Add hello world example

### LINKED ISSUE #10: Publish hello world example
Omaha client is not trivial to use. Publish a minimal but fully functional example illustrating how to use the client lib.

### PR DESCRIPTION
**Pull request by ephemeralriggs**
Add an example how the omaha client lib is used. This example is fully functional and ready to run. It requires the constants SERVICE_URL and APP_ID in main.rs to be filled in when run against a real omaha server. Also include the example in CI runs.
While on it: Fix clippy warnings with recent nightly.

Fixed: #10

### CODE REVIEW

File: examples/hello-world/http_request.rs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        Client, Request, Response,
+    },
+    omaha_client::http_request::{Error, HttpRequest},
+    std::time::Duration,
+};
+
+// This implements the minimal trait requirements for a http request for the omaha state machine.
+// For a detailed implementation, see
+// https://cs.opensource.google/fuchsia/fuchsia/+/main:src/sys/pkg/lib/omaha-client-fuchsia/src/http_request.rs
+
```
Comment (erickt) [inline examples/hello-world/http_request.rs]:
```suggestion

```

File: examples/hello-world/app_set.rs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+#[derive(Debug, Eq, PartialEq, Copy, Clone)]
+pub enum AppIdSource {
+    VbMetadata,
+    ChannelConfig,
+    DefaultEmpty,
+}
+
+#[derive(Debug, Eq, PartialEq, Copy, Clone)]
+pub struct AppMetadata {
+    // The source from which the app's ID was derived.
```
Comment (erickt) [inline examples/hello-world/app_set.rs]:
```suggestion
    /// The source from which the app's ID was derived.
```

File: src/time.rs [hunk 3]
```diff
@@ -64,7 +64,7 @@
 //! When modeling the time required (e.g. timer waits):
 //! * `ComplexTime` - When both wall and monotonic times are always required
 //! * `PartialComplexTime` - When some time (wall, monotonic, or both) is always required, but any
-//! or both will suffice.
+//!   or both will suffice.
```
Comment (erickt) [inline src/time.rs:67]:
```suggestion
//! or both will suffice.
```

**Review by erickt — CHANGES_REQUESTED** [review]
otherwise LGTM

File: src/time.rs (hunk 3 above)
Comment (ephemeralriggs) [reply src/time.rs:67]: Removing the spaces will fail clippy on nightly. This commit adds those spaces just to pet clippy.

**Review by erickt — APPROVED** [review]

**Pull request merged**